Source/Core/DolphinQt2/Config/CheatWarningWidget.cpp

```diff
@@ -0,0 +1,83 @@
+// Copyright 2017 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include "DolphinQt2/Config/CheatWarningWidget.h"
+
+#include <QHBoxLayout>
+#include <QLabel>
+#include <QMessageBox>
+#include <QPixmap>
+#include <QPushButton>
+#include <QStyle>
+#include <QTimer>
+
+#include "Core/ConfigManager.h"
+#include "Core/Core.h"
+#include "DolphinQt2/Settings.h"
+
+CheatWarningWidget::CheatWarningWidget(const std::string& game_id) : m_game_id(game_id)
+{
+  CreateWidgets();
+  ConnectWidgets();
+
+  setHidden(true);
+
+  connect(&Settings::Instance(), &Settings::EnableCheatsChanged, this, &CheatWarningWidget::Update);
```

Comment:
This signals needs to connect to `[this](){ Update(Core::IsRunning()); }` (or just move the `Core::IsRunning` check inside Update), instead of passing along the `cheats_enabled` variable as the `running` argument.